sdk/src/azure/core/az_http_policy.c

```diff
@@ -46,6 +46,11 @@ AZ_NODISCARD az_result az_http_pipeline_policy_apiversion(
 // "-1" below is to account for the null terminator at the end of the string.
 #define _az_TELEMETRY_ID_PREFIX "azsdk-c-"
 #define _az_TELEMETRY_ID_PREFIX_LENGTH (sizeof(_az_TELEMETRY_ID_PREFIX) - 1)
+#define _az_TELEMETRY_COMPONENT_NAME_MAX_LENGTH 40
+#define _az_TELEMETRY_VERSION_MAX_LENGTH (sizeof("12.345.6789-preview.123") - 1)
+#define _az_TELEMETRY_ID_MAX_LENGTH                                                       \
```

Comment:
Roughly what's the value of this? Might be worth putting in a comment. Is it 72?
8 + 40 + 1 + 23 = 72?

Presumably if/when we add things like application id, this will grow by some 24-28 chars, right? Should we add that now, or not worth the effort?

Reply:
I'd avoid to put 72, if the constant may change, so comment will become misleading.

This thing came in unexpected, I don't think there's necessarily enough time to add good `application_id` support. If we're lucky - yes, but should we run into a decision fork, we may not have enough time to think it well through, and I don't want the deadlines to have pressure on the decision making.

I'd propose is to add a public telemetry options struct, with the nullable `application_id` field. That struct will go to all HTTP client options. And there will be telemetry options struct with underscore, that will contain that public struct, plus the `component_id`. We will add another constructor for it that takes two parameters - `component_id` and that new public struct.

But I think it is ok to do it later.
I also can see if we do it now, it might be worth to give it 2-3 days of PR reviews.
I.e. unless you think that it is much more beneficial to add it now, I'd leave it for the future.

Reply:
Yeah, makes sense, we can address it later.